carousel/core/__init__.py

```diff
@@ -279,14 +279,23 @@ def get_parents(bases, parent):
         return [b for b in bases if isinstance(b, parent)]
 
 
-class Field(object):
+class Parameter(dict):
     _attrs = []
 
     def __init__(self, *args, **kwargs):
-        for attr, val in zip(self._attrs, args):
-            setattr(self, attr, val)
+        items = dict(zip(self._attrs, args))
+        extras = {}
         for key, val in kwargs.iteritems():
             if key in self._attrs:
-                setattr(self, key, val)
+                items[key] = val
             else:
+                extras[key] = val
                 LOGGER.warning('This key: "%s" is not an attribute.', key)
+        super(Parameter, self).__init__(items, extras=extras)
```

Comment:
ok, so we're keeping the extra attribute fields entered. It's probably dealt with later but I was wondering what would happen to the attribute fields not entered when instantiating.